ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/BlockSimulator.java

```diff
@@ -135,12 +147,24 @@ public List<BlockSimulationResult> process(
         fillBlockStateCalls(simulationParameter.getBlockStateCalls(), blockHeader);
 
     BlockHeader currentBlockHeader = blockHeader;
+    HashMap<Long, Hash> blockHashCache = new HashMap<>();
```

Comment:
opt: it seems that the number of elements in the `blockHashCache` and `results` is known so your can size them on creation to avoid resizing. 
Use `HashMap::newHashMap` for maps.